### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by imabdulbasit**
Refactor fireblocks client
emitter and signer in separate modules

### CODE REVIEW

File: api/src/events/customer.rs [hunk 1]
```diff
@@ -42,7 +42,12 @@ impl CustomerEventHandler for Processor {
             auto_fuel: Some(false),
         };
 
-        let vault = self.fireblocks.client().create_vault(create_vault).await?;
+        let vault = self
+            .fireblocks
+            .client()
+            .post()
+            .create_vault(create_vault)
```
Comment (kespinola) [inline api/src/events/customer.rs:49]: I thought the post was internal its not very user friendly to call for the post client. Can this be an internal implementation? cleaner for the call to just call create_vault

Comment (kespinola) [reply api/src/events/customer.rs:49]: This API is better if you make it crud based and then the object so .create().vaault()

File: api/src/events/organization.rs [hunk 2]
```diff
@@ -90,6 +95,7 @@ impl OrganizationEventHandler for Processor {
             let vault_asset = self
                 .fireblocks
                 .client()
+                .post()
                 .create_vault_wallet(
                     treasury.vault_id.clone(),
                     asset_type.into(),
```
Comment (kespinola) [inline api/src/events/organization.rs:101]:
```suggestion
                .create()
                .wallet(
                    treasury.vault_id.clone(),
                    asset_type.into(),
```

File: fireblocks/src/client.rs [hunk 3]
```diff
@@ -109,183 +177,341 @@ impl Client {
...
+    /// * Failed to serialize the query filters.
+    /// * Failed to deserialize the paged response.
+    ///
+    /// # Returns
+    ///
+    /// Paged response of vault accounts.
+    pub async fn vaults(&self, filters: QueryVaultAccounts) -> Result<VaultAccountsPagedResponse> {
+        let endpoint = "/v1/vault/accounts_paged";
+        self.send(endpoint, filters).await
+    }
```
Comment (kespinola) [inline fireblocks/src/client.rs:361]: This is a get request yes but in the post builder

File: fireblocks/src/client.rs [hunk 4]
```diff
@@ -109,183 +177,341 @@ impl Client {
...
+    /// * Failed to serialize the request body.
+    /// * Failed to deserialize the created vault account details.
+    ///
+    /// # Returns
+    ///
+    /// Created vault account details.
+    pub async fn create_vault(&self, body: CreateVault) -> Result<VaultAccount> {
+        let endpoint = "/v1/vault/accounts".to_string();
+        self.send(&endpoint, body).await
+    }
```
Comment (kespinola) [inline fireblocks/src/client.rs:382]:
```suggestion
    pub async fn vault(&self, body: CreateVault) -> Result<VaultAccount> {
        let endpoint = "/v1/vault/accounts".to_string();
        self.send(&endpoint, body).await
    }
```

File: fireblocks/src/client.rs [hunk 5]
```diff
@@ -89,11 +126,42 @@ impl Client {
...
 
+    #[must_use]
+    pub fn get(&self) -> GetBuilder {
+        GetBuilder {
+            client: self.clone(),
+        }
+    }
+
+    #[must_use]
+    pub fn post(&self) -> PostBuilder {
```
Comment (kespinola) [inline fireblocks/src/client.rs:137]:
```suggestion
    pub fn create(&self) -> PostBuilder {
```

File: fireblocks/src/client.rs [hunk 6]
```diff
@@ -109,183 +177,341 @@ impl Client {
...
+    /// This function can fail if:
+    ///
+    /// * The POST request fails.
+    /// * Failed to serialize the request body.
+    /// * Failed to deserialize the created vault wallet details.
+    ///
+    /// # Returns
+    ///
+    /// Created vault wallet details.
+    pub async fn create_vault_wallet(
```
Comment (kespinola) [inline fireblocks/src/client.rs:403]:
```suggestion
    pub async fn wallet(
```

**Review by kespinola — CHANGES_REQUESTED** [review]
the wait and sign kinda mess with the predictability of the get and post. I suggest it being internal and keeping the current call signatures.

File: api/src/events/customer.rs (hunk 1 above)
Comment (imabdulbasit) [reply api/src/events/customer.rs]: we are not using any get endpoint so I think we can keep the fireblocks the same

**Pull request merged**